crypto/fipsmodule/digest/digest.c

```diff
@@ -261,11 +275,27 @@ int EVP_DigestInit(EVP_MD_CTX *ctx, const EVP_MD *type) {
 }
 
 int EVP_DigestUpdate(EVP_MD_CTX *ctx, const void *data, size_t len) {
+  if (ctx->digest == NULL) {
+    return 0;
+  }
+
+  if (!SHA3_requested_and_enabled(ctx->digest)) {
+    return 0;
+  }
+
   ctx->digest->update(ctx, data, len);
   return 1;
 }
 
 int EVP_DigestFinal_ex(EVP_MD_CTX *ctx, uint8_t *md_out, unsigned int *size) {
+  if (ctx->digest == NULL) {
```

Comment:
Nit: I guess `SHA3_requested_and_enabled()` could handle this as well. But okay.

Reply:
I thought that checking in the `EVP_` function may be a better option since it will cover cases when other digest `EVP_Update` function is being used without first calling the `EVP_Init`.